utils/S3UtilsMongoClient.js

```diff
@@ -74,6 +74,92 @@ class S3UtilsMongoClient extends MongoClientInterface {
         }
     }
 
+    async updateInflightDeltas(allMetrics, log) {
+        let cursor;
+        try {
+            const bucketKeys = [];
+
+            if (!allMetrics || !Array.isArray(allMetrics) || allMetrics.length === 0) {
+                return allMetrics;
+            }
+
+            const usersBucketCreationDatesMap = await this._getUsersBucketCreationDates(log);
+            Object.keys(usersBucketCreationDatesMap).forEach(bucketEntry => {
+                const bucketName = bucketEntry.split(constants.splitter)[1];
+                const bucketKey = `bucket_${bucketName}_${new Date(usersBucketCreationDatesMap[bucketEntry]).getTime()}`;
+                bucketKeys.push(bucketKey);
+            });
+
+            cursor = await this.getCollection(INFOSTORE).find({
+                _id: { $in: bucketKeys },
```

Comment:
is this filtering relevant?
it opens the possibility of slowing the query, while there will little difference ; and that difference , if any... and it would not change much to have a few more entries anyway, since we traverse the filtered list (allMetrics) anyway...

Reply:
Removed the filtering, it's indeed not relevant: deleted buckets or accounts will not be parsed anyway